avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroEncoder.java

```diff
@@ -46,19 +47,25 @@ public KafkaAvroEncoder(VerifiableProperties props) {
     if (url == null) {
       throw new ConfigException("Missing schema registry url!");
     }
+
     int maxSchemaObject = props.getInt(
-        AbstractKafkaAvroSerDeConfig.MAX_SCHEMAS_PER_SUBJECT_CONFIG,
-        AbstractKafkaAvroSerDeConfig.MAX_SCHEMAS_PER_SUBJECT_DEFAULT);
+            AbstractKafkaAvroSerDeConfig.MAX_SCHEMAS_PER_SUBJECT_CONFIG,
+            AbstractKafkaAvroSerDeConfig.MAX_SCHEMAS_PER_SUBJECT_DEFAULT);
     schemaRegistry = new CachedSchemaRegistryClient(url, maxSchemaObject);
+    enableAutoSchemaRegistration = props.getBoolean(KafkaAvroSerializerConfig.ENABLE_AUTO_SCHEMA_REGISTRATION_CONFIG);
   }
 
   @Override
   public byte[] toBytes(Object object) {
     if (object instanceof IndexedRecord) {
       String subject = ((IndexedRecord) object).getSchema().getName() + "-value";
-      return serializeImpl(subject, object);
+      return serializeImpl(subject, object, enableAutoSchemaRegistration);
     } else {
       throw new SerializationException("Primitive types are not supported yet");
     }
   }
+
+  void setEnableAutoSchemaRegistration(boolean flag) {
```

Comment:
I see you've added setEnableAutoSchemaRegistration to the KafkaAvroSerializer subclasses for testing purposes.

There are also constructors in each of these classes used only for testing (each constructor that takes only SchemaRegistryClient as its argument)
e.g. KafkaAvroEncoder(SchemaRegistryClient ...)

Maybe ditch these setter methods, and instead update these "test constructors" to allow specifying this new parameter. Also, as a general improvement, to the preexisting code, it would be good to add a comment to those constructors to point out they are used exclusively for testing.